src/MongoDB.Bson/Serialization/Conventions/ImmutableTypeClassMapConvention.cs

```diff
@@ -105,5 +105,46 @@ private bool CanWrite(PropertyInfo propertyInfo)
             // CanWrite gets true even if a property has only a private setter
             return propertyInfo.CanWrite && (propertyInfo.SetMethod?.IsPublic ?? false);
         }
+
+        private bool PropertyMatchesSomeCreatorParameter(BsonClassMap classMap, PropertyInfo propertyInfo)
+        {
+            foreach (var creatorMap in classMap.CreatorMaps)
+            {
+                if (creatorMap.MemberInfo is ConstructorInfo constructorInfo)
+                {
+                    if (PropertyMatchesSomeConstructorParameter(constructorInfo))
+                    {
+                        return true;
+                    }
+                }               
+            }
+
+            // also map properties that match some constructor parameter that might be called by a derived class
+            foreach (var constructorInfo in classMap.ClassType.GetTypeInfo().GetConstructors(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance))
+            {
+                if (classMap.ClassType.GetTypeInfo().IsAbstract || constructorInfo.IsFamily)
```

Comment:
We are missing `ctor.IsFamilyOrAssembly` to catch `protected internal` ctors. A `protected internal` ctor returns `false` to `ctor.IsFamily` because it is accessible to more than just derived classes.